### ISSUE TITLE
Align types of class arguments

### ISSUE THREAD

**Issue by densh** [olafurpg/scalafmt#174]
Diff:
```scala
 final case class Focus(
 -  val blocks:     Seq[Block],
 -  val name:       Local,
 -  val params:     Seq[Val.Local],
 -  val insts:      Seq[Inst],
 -  val value:      Val,
 -  val isComplete: Boolean
 +    val blocks: Seq[Block],
 +    val name: Local,
 +    val params: Seq[Val.Local],
 +    val insts: Seq[Inst],
 +    val value: Val,
 +    val isComplete: Boolean
  )
```

I'd like to preserve alignment of types after column.

Using: 0.2.1

**Comment by olafurpg** [comment]
Work on this issue is well ahead on this branch: https://github.com/olafurpg/scalafmt/tree/align-colon

The main difference is that we insert vertical alignment spaces for `NoSplit`, previously it was only for `Space`. The reason I've been reluctant to merge the branch is that it doesn't play so nicely in cases like this

```scala
case class Bar(i: Int) extends B
case object A extends B
```

In this example we have two potential alignment candidates for Bar, the `:` and the `extends`. There is only one alignment candidate for A, the `extends`. The current implementation only aligns the prefix of alignment candidates that match, which in this case is 0 candidates.

**Comment by olafurpg** [comment]
Closed by #441. The caveats above still apply, but those warrant a separate issue.
